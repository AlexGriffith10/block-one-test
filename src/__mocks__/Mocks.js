import React from 'react';
import Enzyme from 'enzyme'

export const dummyProps = [
    {id: 1, timestamp: '05:00', previous: 0, transactions: [1,2,3]},
    {id: 2, timestamp: '05:00', previous: 1, transactions: [1,2,3]},
    {id: 3, timestamp: '05:00', previous: 2, transactions: [1,2,3]},
    {id: 4, timestamp: '05:00', previous: 3, transactions: [1,2,3]},
    {id: 5, timestamp: '05:00', previous: 4, transactions: [1,2,3]},
    {id: 6, timestamp: '05:00', previous: 5, transactions: [1,2,3]},
    {id: 7, timestamp: '05:00', previous: 6, transactions: [1,2,3]},
    {id: 8, timestamp: '05:00', previous: 7, transactions: [1,2,3]},
    {id: 9, timestamp: '05:00', previous: 8, transactions: [1,2,3]},
    {id: 10, timestamp: '05:00', previous: 9, transactions: [1,2,3]}
]