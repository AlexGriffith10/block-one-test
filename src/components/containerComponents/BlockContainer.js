import React, { Component } from 'react'
import BlockItem from '../presentationComponents/BlockItem'

class BlockContainer extends Component {

    render() {
        console.log("my new blocks")
        console.log(this.props.newBlocks)
        return this.props.newBlocks.map((newBlock) => (<BlockItem newBlock = {newBlock} />
            )
        )
    }
}

export default BlockContainer
