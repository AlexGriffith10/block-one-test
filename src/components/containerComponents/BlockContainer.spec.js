import React from  'react';
import {shallow} from 'enzyme';
import BlockContainer from './BlockContainer';
import {dummyProps} from '../../__mocks__/Mocks'

describe("Button Presentation Component", () => {
    test("renders component", () => {
        const wrapper = shallow(<BlockContainer allBlocks={dummyProps}/>);

        expect(wrapper.exists()).toBe(true)
    })
})

describe("snapshot test", () => {
    test("check snapshot against component", () => {
        const wrapper = shallow(<BlockContainer allBlocks={dummyProps} />);

        expect(wrapper).toMatchSnapshot();
    })
})