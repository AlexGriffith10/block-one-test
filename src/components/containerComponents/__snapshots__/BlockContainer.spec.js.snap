// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test check snapshot against component 1`] = `ShallowWrapper {}`;
